roles/jupyterhub/templates/jupyterhub_config.py

```diff
@@ -119,74 +124,95 @@ async def authenticate(self, handler, data):
-    #  self.log.info("In options_form")
-    #  if spawner.user_options['presentation_type']:
-    #    return self.main_options_form # Omit the conda env dropdown since that is chosen per dashboard
 
-    # Display full form including conda env dropdown
-    return ''.join([self.main_options_form, self.conda_options_form])
+
+{% if jupyterhub_qhub_options_form %}
+    # data from form submission is {key: [value]}
+    # we need to convert the formdata to a key value dict
+    def options_from_form(self, data):
+        return {key: value[0] for key, value in data.items()}
+
+    main_options_form = f'''
+    <div class="form-group row">
+    <label for="memory" class="col-2 col-form-label">JupyterLab Memory (GB)</label>
+    <div class="col-10">
+        <input class="form-control" type="number" value="1" id="memory" name="memory">
+    </div>
+    </div>
+    <div class="form-group row">
+    <label for="nprocs" class="col-2 col-form-label">JupyterLab CPUs</label>
+    <div class="col-10">
+        <input class="form-control" type="number" value="1" id="nprocs" name="nprocs">
+    </div>
+    </div>
+    <div class="form-group row">
+    <label for="partition" class="col-2 col-form-label">Slurm Partition</label>
+    <div class="col-10">
+        <select class="form-select" aria-label="Slurm Queue" id="partition" name="partition">
+        <option value="general">general</option>
+    {% for item in groups %}
+    {% if item.startswith('partition-')%}
+        <option value="{{ item[10:] }}">{{ item[10:] }}</option>
+    {% endif %}
+    {% endfor %}
+        </select>
+    </div>
+    </div>
+    '''
+
+    conda_options_form = f'''
+    {% raw %}
+    <div class="form-group row">
+    <label for="conda_environment_prefix" class="col-2 col-form-label">Conda Environment</label>
+    <div class="col-10">
+        <select class="form-select" aria-label="Conda Environment" id="conda_environment_prefix" name="conda_environment_prefix">
+    {''.join([f'<option value="{_[1]}">{_[0]}</option>' for _ in conda_envs_w_packages(jupyterlab_packages)])}
+        </select>
+    </div>
+    </div>
+    {% endraw %}
+    '''
+
+    def options_form(self, spawner):
+        ## Not currently working - idea is to omit conda env from spawner form
+        ## for dashboards, since already selected a conda env
+        #if spawner.orm_spawner.user_options and 'presentation_type' in spawner.orm_spawner.user_options:
+        #  self.log.info("In options_form")
+        #  if spawner.user_options['presentation_type']:
+        #    return self.main_options_form # Omit the conda env dropdown since that is chosen per dashboard
+
+        # Display full form including conda env dropdown
+        return ''.join([self.main_options_form, self.conda_options_form])
 {% endif %}
 
 # Assign Qhub Spawner
 class QHubHPCSpawner(QHubHPCSpawnerBase):
     pass
 
+c.JupyterHub.allow_named_servers = True
+c.JupyterHub.default_url = '/hub/home'
+
```

Comment:
I am not too sure where this came from, can be removed if necessary, but did not affect the instance creation

Reply:
We need to remove anything that isn't required to fix the underlying issue.